### PR TITLE
bugfix on event list page

### LINKED ISSUE #6754: [Events] User cannot interact with 'Search' element on 'Events' page.
**Environment:** Chrome 119.0.6045.200(64bit).
**Reproducible:** always.
**Build found:** last commit from https://github.com/ita-social-projects/GreenCityClient/tree/dev
E.g. "commit 147fa44"

**Preconditions**
Moved to https://www.greencity.social/#/greenCity
Sign-in as user.

**Steps to reproduce**
1. Go to 'Events' page.
2. Click on 'Search' element.
3. Enter the name of the existing event to search.

**Actual result**
Result of the search appeared, but after 5 sec. user have this message - "We didn't find any results matching to this search".
image

**Expected result**
The search element should work correctly and search by the specified parameters.

**Postconditions**
A clear and concise description of actions to be done after bug reproducing. E.g.:
"Delete created item from database using SQL query:
DELETE FROM table1
WHERE id=<item1_id>;"

**User story and test case links**
E.g.: "User story #100
[Test case](https://jira.softserve.academy/browse/100)"

**Labels to be added**
"Bug", Priority ("pri: "), Severity ("severity:"), Type ("UI, "Functional"), "API" (for back-end bugs).

### LINKED ISSUE #6795: [Events] Filter 'Any Time' should be set by default.
**Environment:** Firefox 121.0 (64-bit).
**Reproducible:** always
**Build found:** last commit from https://github.com/ita-social-projects/GreenCityClient/tree/dev
E.g. "commit 147fa44"

**Preconditions**
Moved to https://www.greencity.social/#/greenCity

**Steps to reproduce**
1. Go to 'Events' page.
2. Click on 'Event time' dropdown.

**Actual result**
Filter 'Any Time' dont set by default in 'Event time drop down'.

**Expected result**
Filter 'Any Time' should be set by default in 'Event time drop down'.

**Postconditions**
A clear and concise description of actions to be done after bug reproducing. E.g.:
"Delete created item from database using SQL query:
DELETE FROM table1
WHERE id=<item1_id>;"

**User story and test case links**
E.g.: "User story #100
[Test case](https://jira.softserve.academy/browse/100)"

**Labels to be added**
"Bug", Priority ("pri: "), Severity ("severity:"), Type ("UI, "Functional"), "API" (for back-end bugs).

### LINKED ISSUE #6796: [Events] User cannot find saved Events.
**Environment:** Firefox 121.0 (64-bit).
**Reproducible:** always.
**Build found:** last commit from https://github.com/ita-social-projects/GreenCityClient/tree/dev
E.g. "commit 147fa44"

**Preconditions**
Moved to https://www.greencity.social/#/greenCity
User need have saved Events.

**Steps to reproduce**
1. Go to 'Events' page.
2. Click on 'Saved Events' bttn.

**Actual result**
'Saved Events' bttn disabled.

**Expected result**
'Saved Events' bttn should be enabled.
image

**Postconditions**
A clear and concise description of actions to be done after bug reproducing. E.g.:
"Delete created item from database using SQL query:
DELETE FROM table1
WHERE id=<item1_id>;"

**User story and test case links**
E.g.: "User story #100
[Test case](https://jira.softserve.academy/browse/100)"

**Labels to be added**
"Bug", Priority ("pri: "), Severity ("severity:"), Type ("UI, "Functional"), "API" (for back-end bugs).

### PR DESCRIPTION
**Pull request by igorv404**
# GreenCityClient PR
[#6754](https://github.com/ita-social-projects/GreenCity/issues/6754)
[#6796](https://github.com/ita-social-projects/GreenCity/issues/6796)
[#6795](https://github.com/ita-social-projects/GreenCity/issues/6795)

## Summary Of Changes :fire:
Implemented search event method, implemented get favourite events method and cleaned up code

## How to test :clipboard:
Repeat actions in following request: [#6754](https://github.com/ita-social-projects/GreenCity/issues/6754), [#6796](https://github.com/ita-social-projects/GreenCity/issues/6796), [#6795](https://github.com/ita-social-projects/GreenCity/issues/6795)
# PR Checklist Forms

- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: src/app/main/component/events/components/events-list/events-list.component.html [hunk 1]
```diff
@@ -116,13 +116,19 @@
...
       infiniteScroll
       [infiniteScrollDistance]="0"
       [infiniteScrollThrottle]="500"
-      (scrolled)="onScroll()"
+      (scrolled)="getEvents()"
     ></app-spinner>
-    <p class="end-page" *ngIf="!elementsArePresent">{{ 'homepage.events.no-event-left' | translate }}</p>
-    <p class="end-page" *ngIf="noEventsMatch">{{ 'homepage.events.no-matching-left' | translate }}</p>
+    <app-spinner
+      *ngIf="isLoading && eventsList.length === 0"
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.html]: isLoading && !eventsList.length

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 2]
```diff
@@ -213,23 +201,25 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
-          filter: this.eventFilterCriteria
-        })
-      );
-    } else {
-      this.scroll = false;
-      this.elementsArePresent = false;
+  public getEvents(): void {
+    if (this.hasNextPage) {
+      const searchTitle = this.searchEventControl.value.trim();
+      if (searchTitle.length === 0) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts]: the same - if (!searchTitle.length)

File: src/app/main/component/events/components/events-list/events-list.component.html (hunk 1 above)
Comment (igorv404) [reply src/app/main/component/events/components/events-list/events-list.component.html]: Done

File: src/app/main/component/events/components/events-list/events-list.component.ts (hunk 2 above)
Comment (igorv404) [reply src/app/main/component/events/components/events-list/events-list.component.ts]: Done

**Review by hnativlyubomyr — APPROVED** [review]

**Review by sobkiv — APPROVED** [review]

**Review by dmytro-kryvyi — APPROVED** [review]

**Review by OksanaFilipchuk — APPROVED** [review]

**Pull request merged**